### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by SuperQ**
* Update Go build to 1.18.
* Update Go modules for 1.17 format.
* Update Go modules versions.
* Update upstream Prometheus template files.
* Enable dependabot.

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Review by sysadmind — APPROVED** [review]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ jobs:
 
     steps:
     - prometheus/setup_environment
-    - run: sudo apt-get -y install build-essential libsnmp-dev
+    - run: apt-get update
```
Comment (RichiH) [inline .circleci/config.yml:25]: Why no update any more?

File: .circleci/config.yml [hunk 2]
```diff
@@ -38,7 +39,8 @@ jobs:
     steps:
     - checkout:
         path: ~/project
-    - run: sudo apt-get -y install build-essential diffutils libsnmp-dev p7zip-full
+    - run: apt-get update
```
Comment (RichiH) [inline .circleci/config.yml:42]: Why no update any more?

**Review by RichiH — APPROVED** [review]
LGTM, just two questions inlined

File: .circleci/config.yml (hunk 1 above)
Comment (SuperQ) [reply .circleci/config.yml:25]: Change to the Prometheus golang-builder image. Doesn't include pre-populated apt package cache.

**Pull request merged**